quic/s2n-quic-core/src/stream/limits.rs

```diff
@@ -27,7 +35,75 @@ impl Default for Limits {
 
 impl Limits {
     pub const RECOMMENDED: Self = Self {
-        max_send_buffer_size: DEFAULT_STREAM_MAX_SEND_BUFFER_SIZE,
-        max_open_local_unidirectional_streams: InitialMaxStreamsUni::RECOMMENDED.as_varint(),
+        max_send_buffer_size: LocalMaxSendBufferSize::RECOMMENDED,
+        max_open_local_unidirectional_streams: LocalUniDirectionalLimit::RECOMMENDED,
+        max_open_local_bidirectional_streams: LocalBiDirectionalLimit::RECOMMENDED,
+    };
+}
+
+macro_rules! local_limits {
+    ($name:ident($encodable_type:ty)) => {
+        #[derive(Debug, Copy, Clone, PartialEq, Eq)]
+        pub struct $name($encodable_type);
     };
 }
+
+// VarInt specific functionality
+macro_rules! varint_local_limits {
+    ($name:ident($encodable_type:ty)) => {
+        local_limits!($name($encodable_type));
+
+        impl $name {
+            pub const fn as_varint(&self) -> VarInt {
+                self.0
+            }
+        }
+
+        impl TryFrom<u64> for $name {
+            type Error = ValidationError;
+
+            fn try_from(value: u64) -> Result<Self, Self::Error> {
+                let value = VarInt::new(value)?;
+                Ok(Self(value))
+            }
+        }
+    };
+}
+
+local_limits!(LocalMaxSendBufferSize(u32));
+
+impl LocalMaxSendBufferSize {
+    pub const RECOMMENDED: Self = Self(DEFAULT_STREAM_MAX_SEND_BUFFER_SIZE);
+
+    pub fn as_u32(self) -> u32 {
+        self.0
+    }
+}
+
+impl TryFrom<u32> for LocalMaxSendBufferSize {
+    type Error = ValidationError;
+
+    fn try_from(value: u32) -> Result<Self, Self::Error> {
+        Ok(Self(value))
+    }
+}
+
+varint_local_limits!(LocalUniDirectionalLimit(VarInt));
+
+impl LocalUniDirectionalLimit {
+    pub const RECOMMENDED: Self = Self(InitialMaxStreamsUni::RECOMMENDED.as_varint());
+}
+
+varint_local_limits!(LocalBiDirectionalLimit(VarInt));
+
+impl LocalBiDirectionalLimit {
+    pub const RECOMMENDED: Self = Self(InitialMaxStreamsBidi::RECOMMENDED.as_varint());
+}
+
+// To maintain backwards API compatibility we need to convert from
+// `max_open_bidirectional_streams` to `max_open_local_bidirectional_streams`
```

Comment:
Can you explain this comment a bit more? I think the API backwards compatibility is not affected, since the APIs just deal with u64s, but you are talking more about maintaining existing behavior?

Reply:
Yep I meant existing behavior.. I think the comment currently is kinda confusing.

The exact behavior I am aiming for is in the written in the PR details at `https://github.com/aws/s2n-quic/pull/1394`.

Reply:
I dont think the comment is useful here and once we introduce the new API behavior it will be better to document it there. Removing this comment.